Assets/GothicVR/Scripts/Player/Running/MovementTypeController.cs

```diff
@@ -0,0 +1,203 @@
+            var rightClicked = GetButtonClicked(InputDeviceCharacteristics.Right, CommonUsages.gripButton);
+            var otherMovementClicked = GetButtonClicked(InputDeviceCharacteristics.Left, CommonUsages.primary2DAxis);
+
+            // Exclude normal walking from ArmLocomotion.
+            if (!leftClicked || !rightClicked || otherMovementClicked)
+            {
+                return;
+            }
+
+            Vector3 leftHandVelocity = leftHand.transform.position - previousPosLeft;
+            Vector3 rightHandVelocity = rightHand.transform.position - previousPosRight;
+            float totalVelocity = leftHandVelocity.magnitude + rightHandVelocity.magnitude;
+
+            if (totalVelocity >= MIN_VELOCITY_ARM_SWING_ACTIVATE)
+            {
+                if (!_armSwingActive)
+                {
+                    SetRunningSpeed(running_speed);
+                    _armSwingActive = true;
+                }
+
+                _armSwingActive = true;
+                var direction = Camera.main.transform.forward;
+                character.Move(running_speed * Time.deltaTime * Vector3.ProjectOnPlane(direction, Vector3.up));
+            }
+            else if (_armSwingActive)
+            {
+                SetWalkingSpeed(walking_speed);
+                _armSwingActive = false;
+            }
+
+            SetPreviousArmPosition();
+        }
+
+        private void SetPreviousArmPosition()
+        {
+            previousPosLeft = leftHand.transform.position;
+            previousPosRight = rightHand.transform.position;
+        }
+
+        // @see XR Input mappings https://docs.unity3d.com/Manual/xr_input.html
+        private void SetMovementSpeed()
+        {
+            var clicked = GetButtonClicked(InputDeviceCharacteristics.Left, CommonUsages.primary2DAxisClick);
+
+            // nothing's clicked?
+            if (!clicked)
+            {
+                _runClickHandled = false;
+                return;
+            }
+
+            // Click was already handled?
+            if (_runClickHandled)
+            {
+                return;
+            }
+
+            // Toggle speed on every button press.
+
+            if (_locomotionProvider.moveSpeed == walking_speed)
+            {
+                SetRunningSpeed(running_speed);
+            }
+            else
+            {
+                SetWalkingSpeed(walking_speed);
+            }
+
+            // Do not execute speed change for the duration of this click again.
+            _runClickHandled = true;
+        }
+
+        private void SetWalkingSpeed(float speed)
+        {
+            _locomotionProvider.moveSpeed = speed;
+            //runningCanvas.enabled = false;
+        }
+
+        private void SetRunningSpeed(float speed)
+        {
+            _locomotionProvider.moveSpeed = speed;
+            //runningCanvas.enabled = true;
+        }
```

Comment:
No need to have two functions that do the exact same thing

```suggestion
        private void SetSpeed(float speed)
        {
            _locomotionProvider.moveSpeed = speed;
        }
```